src/dotnet/MsbuildProject.cs

```diff
@@ -166,12 +174,97 @@ public static void EnsureAllReferencesExist(List<string> references)
             }
         }
 
+        public IEnumerable<NuGetFramework> GetTargetFrameworks()
+        {
+            if (_cachedTfms != null)
+            {
+                return _cachedTfms;
+            }
+
+            var project = GetEvaluatedProject();
+
+            var properties = project.AllEvaluatedProperties
```

Comment:
Why are we diverging from `dotnet run` here? There is already a nice helper method for extracting a Property from a ProjectInstance:
https://github.com/dotnet/cli/blob/706b8a802d49be079fef570b354a70056aa2ad6f/src/dotnet/commands/dotnet-run/RunCommand.cs#L90

Please re-use the same approach in both places.

Note, the current code may be confusing due to https://github.com/dotnet/cli/issues/4977

Edit: If the method invoked by Run is concretely on ProjectInstance, please create a ProjectInstanceExtensions.cs and move this logic there. Let's maintain similar patterns.